File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SpecialSubject/Top/reverse_7.java
package algo.tzashinorpu.SpecialSubject.Top;

public class reverse_7 {
    public int reverse(int x) {
        long res = 0;
        // 12345
        while (x != 0) {
            int tmp = x % 10;
            res = (res * 10) + tmp;
            x = x / 10;
            if (res > Integer.MAX_VALUE || res < Integer.MIN_VALUE) return 0;
        }
        return (int) res;
    }
}

File: Cpp/Algorithm/src/SpecialSubject/Reverse_7.cpp
#include "ss.h"
class Reverse_7
{
public:
	Reverse_7();
	~Reverse_7();
	int reverse(int x) {
		long res = 0;
		while (x != 0)
		{
			int c = x % 10;
			res = res * 10 + c;
			x = x / 10;
			if (res > INT_MAX || res < INT_MIN) return 0;
		}
		return res;
	}
private:

};

Reverse_7::Reverse_7()
{
}

Reverse_7::~Reverse_7()
{
}